R/trimData.R

```diff
@@ -28,23 +28,39 @@ trimData <- function(data, settings, chart="edish"){
 
   # Add items in vectors to list individually
   dataVectorKeys <- allKeys %>% filter(.data$setting_type =="vector") %>% pull(.data$text_key) %>% textKeysToList()
+  
+  vectorVars <- list()
+  
   for(key in dataVectorKeys){
     current<-getSettingValue(key, settings=settings)
+
     if (length(current) > 0 ) {
-      for (i in 1:length(current)){
-        newKey <- key
-        newKey[[1+length(newKey)]]<-i
-        sub <- current[[i]]
-        if(typeof(sub)=="list"){
-          newKey[[1+length(newKey)]]<-"value_col"
+      # handle vectors separately
```

Comment:
the true vectors aren't going to have a separate key for each value - so handling it differently.